### PR TITLE
Wrong DataItems groups treatment

### PR DESCRIPTION
**Pull request by CaioCordeiro**
This patch treats wrong DataItems cases.

Changes:
	- Create check_valid_groups function;
	- Refactor on the get_data_item_groups function;
	- Add NULL treatment on set_data_items function;

### CODE REVIEW

File: src/storage.c [hunk 1]
```diff
@@ -102,6 +103,34 @@ static int save_settings(int fd, struct l_settings *settings)
...
+	if (!regexec(&preg, all_groups[index], 0, (regmatch_t *)NULL, 0) == 0) {
+		l_error("Invalid DataItem group: %s", all_groups[index]);
+		err = -EINVAL;
+	}
+
+	for (i = index + 1; all_groups[i] != NULL; i++) {
+		repeated_check = strcmp(all_groups[i], all_groups[index]);
+		if (repeated_check == 0) {
+			l_error("Repeated DataItem group: %s", all_groups[i]);
+			err = -EINVAL;
```
Comment (agms-cesar) [inline src/storage.c:124]: you can break this for here if an err already occured

File: src/storage.c [hunk 2]
```diff
@@ -102,6 +103,34 @@ static int save_settings(int fd, struct l_settings *settings)
...
+	int repeated_check;
+	regex_t preg;
+	int i;
+	int err;
+
+	regcomp(&preg, "^DataItem_[0-9]+$", REG_EXTENDED);
+
+	if (!regexec(&preg, all_groups[index], 0, (regmatch_t *)NULL, 0) == 0) {
+		l_error("Invalid DataItem group: %s", all_groups[index]);
+		err = -EINVAL;
```
Comment (agms-cesar) [inline src/storage.c:117]: You can immediately return the error here to avoid entering the for loop.

File: src/storage.c (hunk 1 above)
Comment (CaioCordeiro) [reply src/storage.c]: Done!

File: src/storage.c (hunk 2 above)
Comment (CaioCordeiro) [reply src/storage.c]: Done!

**Review by agms-cesar — APPROVED** [review]

**Review by vitbaq — APPROVED** [review]

**Pull request merged**